lib/python/rosie/suite_id.py

```diff
@@ -143,9 +143,10 @@ def get_latest(cls, prefix=None):
                 if i == 0:
                     return None
                 raise SuiteIdLatestError(prefix)
-            dirs = [line for line in out.splitlines() if line.endswith(b"/")]
```

Comment:
Doesn't necessarily mean that they are wrong though. If @sadielbartholomew  has changed them further upstream this feels like a good call.

Reply:
In general, I have only made changes to Tim's porting work, in practice almost always relating to the handling of strings, if:

    I have observed any of the new, wider set of ported (non-skipped) tests failing with stdout & stderr suggesting some issue resulting ultimately from trademark Python version related issues, making logical tweaks to fix those tests; or
    The ported code, especially with respect to encoding or decoding, could be simplified &/or made cleaner:
        (as Tim indicated) the de- or en-coding is still done as before, but has been moved further upstream or downstream; or
        (as in this case) by converting certain variables (such as dirs here) that are only used as an intermediate to some other variable, to strings from bytes or vice versa.

Most of these discrete changes should therefore be necessary, but I will go through & do a sanity check on all such changes to make sure I am not changing the string-handling behaviour. Thanks for raising this.

Reply:
In general, I have only made changes to Tim's porting work, in practice almost always relating to the handling of strings, if:

1. I have observed any of the new, wider set of ported (non-skipped) tests failing with stdout & stderr suggesting some issue resulting ultimately from trademark Python version related issues, making logical tweaks to fix those tests; or
2. The ported code, especially with respect to encoding or decoding, could be simplified &/or made cleaner:
       * (as Tim indicated) the de- or en-coding is still done as before, but has been moved further upstream or downstream; or
       * (as in this case) by converting certain variables (such as dirs here) that are only used as an intermediate to some other variable, to strings from bytes or vice versa.

Most of these discrete changes should therefore be necessary, but I will go through & do a sanity check on all such changes to make sure I am not changing the string-handling behaviour. Thanks for raising this.

Reply:
In the cases in this particular file, to provide quick reassurance, I've wrote a small script you could run:

```python
import sys

out = bytes(sys.argv[1].encode())
dir_url= str(sys.argv[2])

###------ Before
old_dirs = [line for line in out.splitlines() if line.endswith(b"/")]
old_dir_url = dir_url + "/" + sorted(old_dirs)[-1].rstrip(b"/").decode()

###------- After
new_dirs = [line for line in out.decode().splitlines() if line.endswith("/")]
new_dir_url = dir_url + "/" + sorted(new_dirs)[-1].rstrip("/")

print("intermediate dirs:", old_dirs, new_dirs)  # might differ but it is irrelevant
print("final dir_url:", old_dir_url, new_dir_url)  # need to be the same; no change
```

so run any (valid) example for the file ``out`` & string ``dir_url``, e.g:

```console
$ python tester_temp.py "orange\n/yellow\n/green/" "blue"
intermediate 'dirs': [b'orange\\n/yellow\\n/green/'] ['orange\\n/yellow\\n/green/']
final 'dir_url': blue/orange\n/yellow\n/green blue/orange\n/yellow\n/green

```

to see the same ``dir_url``, even if the ``dirs`` changes from byte to string (though it is not used elsewhere so that is not an issue).

Reply:
In fact, then the only changes here to Tim's porting work are those in ``rosie/svn_post_commit.py``, which were required to fix the associated tests (``t/rosa-db-create`` & ``t/rosa-svn-post-commit``) which were otherwise failing in several ways.